content/docs/migration_1.0/Pico.md

```diff
@@ -0,0 +1,263 @@
+
+We have a serialization function to encode the key/pair value but nothing prevents use to serialize an another type as attachment.
+
+- Zenoh 0.11.x
+
+```c
+char *value = "Some data to publish on Zenoh";
+
+z_put_options_t options = z_put_options_default();
+z_owned_bytes_map_t map = z_bytes_map_new();
+z_bytes_map_insert_by_alias(&map, _z_bytes_wrap((uint8_t *)"test", 2), _z_bytes_wrap((uint8_t *)"attachement", 5));
+options.attachment = z_bytes_map_as_attachment(&map);
+
+if (z_put(z_loan(session), z_keyexpr(ke), (const uint8_t *)value, strlen(value), NULL) < 0) {
+    return -1;
+}
+
+if (z_put(z_loan(s), z_keyexpr(keyexpr), (const uint8_t *)value, strlen(value), &options) < 0) {
+    return -1;
+}
+
+z_bytes_map_drop(&map);
+```
+
+- Zenoh 1.0.0
+
+```c
+
+char *value = "Some data to publish on Zenoh";
+z_owned_bytes_t payload;
+z_bytes_serialize_from_str(&payload, value);
+
+// It's up to the user to create the item serializer function
+z_publisher_put_options_t options;
+z_publisher_put_options_default(&options);
+z_owned_bytes_t attachment;
+z_bytes_serialize_from_iter(&attachment, create_attachment_iter, (void *)&ctx);
+options.attachment = z_move(attachment);
+
+if (z_put(z_loan(session), z_loan(ke), z_move(payload), &options) < 0) {
+  return -1;
+}
+```
+
+## Encoding
+
+How encoding is handled has been reworked:  While before you would use an enum id and a value, now you simply need to register your encoding metadata from a string with `z_encoding_from_str`. 
+
+We have a set of constants to do some wire-level optimization. To use this, you need your string should follow the format: "<constant>;<optional additional data>"
+
+- Zenoh 0.11.x
+
+```c
+char *value = "Some data to publish on Zenoh";
+z_put_options_t options = z_put_options_default();
+options.encoding = z_encoding(Z_ENCODING_PREFIX_TEXT_PLAIN, "utf8");
+
+if (z_put(z_loan(session), z_keyexpr(ke), (const uint8_t *)value, strlen(value), &options) < 0) {
+    return -1;
+}
+```
+
+- Zenoh 1.0.0
+
+```c
+char *value = "Some data to publish on Zenoh";
+z_owned_bytes_t payload;
+z_bytes_serialize_from_str(&payload, value);
+
+z_publisher_put_options_t options;
+z_publisher_put_options_default(&options);
+z_encoding_from_str(&encoding, "text/plain;utf8");
+options.encoding = z_move(encoding);
+
+if (z_put(z_loan(session), z_loan(ke), z_move(payload), &options) < 0) {
+  return -1;
+}
+```
+
+## Timestamps
+
+We now tie generating a timestamp to a Zenoh session, with the timestamp inheriting the `ZenohID` of the session.
+
+- Zenoh 0.11.x
+
+```c
+// Didn't exist in pico
+```
+
+- Zenoh 1.0.0
+
+```c
+z_timestamp_t ts;
+z_timestamp_new(&ts, z_loan(s));
+options.timestamp = &ts;
+z_publisher_put(z_loan(pub), z_move(payload), &options);
+```
+
+## Pull subscribers / channels
+
+Pull subscribers were removed and is replaced by FIFO or ring buffers
+
+## Interests
+
+We have added interests to the protocol. This allows a few things:
```

Comment:
Code examples like with the others sections would be nice

Reply:
@jean-roland Could we add an example here ?

Reply:
Alright but there is no api-level code, it's a protocol optimisation thing.